rest/handler.go

```diff
@@ -725,26 +725,35 @@ func (h *handler) checkAuth(dbCtx *db.DatabaseContext) (err error) {
 		return nil
 	}
 
+	auditFields := make(base.AuditFields)
+
 	// Record Auth stats
 	defer func(t time.Time) {
 		delta := time.Since(t).Nanoseconds()
 		dbCtx.DbStats.Security().TotalAuthTime.Add(delta)
 		if err != nil {
 			dbCtx.DbStats.Security().AuthFailedCount.Add(1)
+			if errors.Is(err, ErrInvalidLogin) {
```

Comment:
do we want to log on other errors here? It seems like this would be important.

I'm not sure what the other errors would be - we could also add another field for error type

Reply:
I think other errors fall under operational logging and metrics more than auditing known events, but I could maybe be convinced otherwise?

Reply:
https://github.com/couchbase/sync_gateway/blob/9737474c05067d3eeee6ed42f528b0f2160e7748/rest/handler.go#L818 seems like a niche error - disabled guest user? I'm actually not sure if it can fall into this pathway.

The argument for putting into the defer is that `checkAuth` is really hard to read and it's not clear if we've always wrapped the error correctly.

I think if you pass in an invalid cookie, you'll get an error that is not `ErrInvalidLogin` https://github.com/couchbase/sync_gateway/blob/9737474c05067d3eeee6ed42f528b0f2160e7748/auth/session.go#L42 but is also a 401 error.

I didn't look more exhaustively through the function.